src/EnergyPlus/SolarShading.cc

```diff
@@ -10441,10 +10441,10 @@ void SkyDifSolarShading(EnergyPlusData &state)
 
         if (state.dataSurface->SurfHasSurroundingSurfProperties(SurfNum)) {
             SrdSurfsNum = state.dataSurface->SurfSurroundingSurfacesNum(SurfNum);
-            if (state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).SkyViewFactor != -1) {
+            if (state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).SkyViewFactor != 0) {
                 state.dataSurface->Surface(SurfNum).ViewFactorSkyIR *= state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).SkyViewFactor;
             }
-            if (state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).GroundViewFactor != -1) {
+            if (state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).GroundViewFactor != 0) {
```

Comment:
Sky and ground view factors are no more initialized to -1.0, so needed these logic changes as well.

Reply:
I think you misunderstood my comment. Do the addition without the if. It’s faster to add 0 than it is to test if something is not 0 and only add in that case.

Reply:
I did I understand your comment but there is a reason that this check is needed because it is overriding an existing value. I am actually looking a way to delete this view factor reset.

Reply:
The view factors update in `SolarShading.cc` module above are hit prior to `state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).GroundViewFactor` and `state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).SkyViewFactor` variables are reset due to the current calling order. Thus, the calling point of the new function InitSurfacePropertyViewFactors() should move up before calling `SkyDifSolarShading()`, preferably in `InitHeatBalance()`. Quick test shows diffs will be introduced but it is justifiable, I think. I will get rid of the view factor update from `SolarShading.cc`  and see it s impact.